File: app/src/main/java/com/lianpos/devfoucs/contacts/adapter/CityAdapter.java
package com.lianpos.devfoucs.contacts.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.lianpos.activity.R;
import com.lianpos.devfoucs.contacts.model.CityBean;
import com.lianpos.devfoucs.homepage.activity.IWantBillingActivity;
import com.lianpos.devfoucs.linkman.ui.LinkManActivity;
import com.lianpos.devfoucs.login.activity.LoginActivity;
import com.lianpos.devfoucs.login.activity.RegisterActivity;

import java.util.List;

/**
 * Created by wangshuai .
 * Date: 17/11/02
 */

public class CityAdapter extends RecyclerView.Adapter<CityAdapter.ViewHolder> {
    protected Context mContext;
    protected List<CityBean> mDatas;
    protected LayoutInflater mInflater;
    public OnRecyclerViewLongItemClickListener mOnLongItemClickListener = null;//长按

    public CityAdapter(Context mContext, List<CityBean> mDatas) {
        this.mContext = mContext;
        this.mDatas = mDatas;
        mInflater = LayoutInflater.from(mContext);
    }

    public List<CityBean> getDatas() {
        return mDatas;
    }

    public CityAdapter setDatas(List<CityBean> datas) {
        mDatas = datas;
        return this;
    }

    @Override
    public CityAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(mInflater.inflate(R.layout.item_city_link, parent, false));
    }

    @Override
    public void onBindViewHolder(final CityAdapter.ViewHolder holder, final int position) {
        final CityBean cityBean = mDatas.get(position);
        holder.tvCity.setText(cityBean.getCity());
        holder.content.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
            }
        });
        holder.content.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                if (mOnLongItemClickListener != null) {
                    mOnLongItemClickListener.onLongItemClick(v, position);
                }
                return true;
            }
        });
        holder.avatar.setText("18842535353");
        holder.tvSupermarket.setText("十七超市");
    }

    @Override
    public int getItemCount() {
        return mDatas != null ? mDatas.size() : 0;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvCity;
        TextView avatar;
        TextView tvSupermarket;
        View content;

        public ViewHolder(View itemView) {
            super(itemView);
            tvCity = (TextView) itemView.findViewById(R.id.tvCity);
            tvSupermarket = (TextView) itemView.findViewById(R.id.tvSupermarket);
            avatar = (TextView) itemView.findViewById(R.id.tvPhone);
            content = itemView.findViewById(R.id.content);
        }
    }

    public interface OnRecyclerViewLongItemClickListener {
        void onLongItemClick(View view, int position);
    }

    public void setOnLongItemClickListener(OnRecyclerViewLongItemClickListener listener) {
        this.mOnLongItemClickListener = listener;
    }
}

File: app/src/main/java/com/lianpos/devfoucs/contacts/adapter/CityAdapter.kt
package com.lianpos.devfoucs.contacts.adapter

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

import com.lianpos.activity.R
import com.lianpos.devfoucs.contacts.model.CityBean
import com.lianpos.devfoucs.homepage.activity.IWantBillingActivity
import com.lianpos.devfoucs.linkman.ui.LinkManActivity
import com.lianpos.devfoucs.login.activity.LoginActivity
import com.lianpos.devfoucs.login.activity.RegisterActivity

/**
 * Created by wangshuai .
 * Date: 17/11/02
 */

open class CityAdapter(protected var mContext: Context, protected var mDatas: List<CityBean>?) : RecyclerView.Adapter<CityAdapter.ViewHolder>() {
    protected var mInflater: LayoutInflater
    var mOnLongItemClickListener: OnRecyclerViewLongItemClickListener? = null//长按

    init {
        mInflater = LayoutInflater.from(mContext)
    }

    fun getDatas(): List<CityBean>? {
        return mDatas
    }

    fun setDatas(datas: List<CityBean>): CityAdapter {
        mDatas = datas
        return this
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CityAdapter.ViewHolder {
        return ViewHolder(mInflater.inflate(R.layout.item_city_link, parent, false))
    }

    override fun onBindViewHolder(holder: CityAdapter.ViewHolder, position: Int) {
        val cityBean = mDatas!![position]
        holder.tvCity.text = cityBean.city
        holder.content.setOnClickListener { }
        holder.content.setOnLongClickListener { v ->
            if (mOnLongItemClickListener != null) {
                mOnLongItemClickListener!!.onLongItemClick(v, position)
            }
            true
        }
        holder.avatar.text = "18842535353"
        holder.tvSupermarket.text = "十七超市"
    }

    override fun getItemCount(): Int {
        return if (mDatas != null) mDatas!!.size else 0
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var tvCity: TextView
        internal var avatar: TextView
        internal var tvSupermarket: TextView
        internal var content: View

        init {
            tvCity = itemView.findViewById(R.id.tvCity) as TextView
            tvSupermarket = itemView.findViewById(R.id.tvSupermarket) as TextView
            avatar = itemView.findViewById(R.id.tvPhone) as TextView
            content = itemView.findViewById(R.id.content)
        }
    }

    interface OnRecyclerViewLongItemClickListener {
        fun onLongItemClick(view: View, position: Int)
    }

    fun setOnLongItemClickListener(listener: OnRecyclerViewLongItemClickListener) {
        this.mOnLongItemClickListener = listener
    }
}
